maritalk/model.py

```diff
@@ -40,21 +46,29 @@ def generate(
             stopping_tokens (`List`, *optional*):
                 A list of tokens to use as a stop criteria.
         """
-        if isinstance(messages, str):
-            messages = [{"role": "user", "content": messages}]
 
-        if not (
-            isinstance(messages, list)
-            and len(messages) > 0
-            and isinstance(messages[0], dict)
-        ):
-            raise TypeError(
-                "Invalid `messages` argument format. It's expected to contain a single `str` or a list of dictionaries containing `role` and `content` fields."
-            )
+        if chat_mode:
+            if not (
+                isinstance(messages, str)
+                or (
+                    isinstance(messages, list)
+                    and len(messages) > 0
+                    and isinstance(messages[0], dict)
+                    )
+            ):
+                raise TypeError(
```

Comment:
Currently, chat_mode=True and messages=string, is not supported in the API and will return an error.

we have these options:
- return an error in this case in the lib
- transform the str in the [{"role": "user", "content": messages}] format in the lib
- change the API to change the str in the [{"role": "user", "content": messages}] format.

Reply:
The PR #8 in "maritalk" repo already implements the 3rd option: change the API to change the str in the [{"role": "user", "content": messages}] format.